src/main/kotlin/no/nav/syfo/oppfolgingstilfelle/bit/kafka/KafkaSyketilfelleBitTask.kt

```diff
@@ -0,0 +1,105 @@
+import no.nav.syfo.oppfolgingstilfelle.database.NoElementInsertedException
+import no.nav.syfo.util.configuredJacksonMapper
+import no.nav.syfo.util.kafkaCallId
+import org.apache.kafka.clients.consumer.ConsumerRecords
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.time.Duration
+
+private val log: Logger = LoggerFactory.getLogger("no.nav.syfo")
+
+const val SYKETILFELLEBIT_TOPIC = "flex.syketilfellebiter"
+
+fun launchKafkaTaskSyketilfelleBit(
+    applicationState: ApplicationState,
+    applicationEnvironmentKafka: ApplicationEnvironmentKafka,
+    database: DatabaseInterface,
+) {
+    launchBackgroundTask(applicationState = applicationState) {
+        blockingApplicationLogicSyketilfelleBit(
+            applicationState = applicationState,
+            applicationEnvironmentKafka = applicationEnvironmentKafka,
+            database = database,
+        )
+    }
+}
+
+fun blockingApplicationLogicSyketilfelleBit(
+    applicationState: ApplicationState,
+    applicationEnvironmentKafka: ApplicationEnvironmentKafka,
+    database: DatabaseInterface,
+) {
+    log.info("Setting up kafka consumer SyketilfelleBit")
+
+    val consumerProperties = kafkaSyketilfelleBitConsumerConfig(applicationEnvironmentKafka)
+    val kafkaConsumerSyketilfelleBit = KafkaConsumer<String, String>(consumerProperties)
+
+    kafkaConsumerSyketilfelleBit.subscribe(
+        listOf(SYKETILFELLEBIT_TOPIC)
+    )
+    while (applicationState.ready) {
+        pollAndProcessSyketilfelleBit(
+            database = database,
+            kafkaConsumerSyketilfelleBit = kafkaConsumerSyketilfelleBit,
+        )
+    }
+}
+
+fun pollAndProcessSyketilfelleBit(
+    database: DatabaseInterface,
+    kafkaConsumerSyketilfelleBit: KafkaConsumer<String, String>,
+) {
+    val records = kafkaConsumerSyketilfelleBit.poll(Duration.ofMillis(1000))
+    if (records.count() > 0) {
+        createAndStoreFromSyketilfelleBitRecords(
+            consumerRecords = records,
+            database = database,
+        )
+        kafkaConsumerSyketilfelleBit.commitSync()
+    }
+}
+
+fun createAndStoreFromSyketilfelleBitRecords(
+    consumerRecords: ConsumerRecords<String, String>,
+    database: DatabaseInterface,
+) {
+    database.connection.use { connection ->
+        consumerRecords.forEach { consumerRecord ->
+            val callId = kafkaCallId()
+            if (consumerRecord.value() == null) {
+                log.error("Value of ConsumerRecord is null, most probably due to a tombstone. Contact the owner of the topic if an error is suspected. key=${consumerRecord.key()} from topic: ${consumerRecord.topic()}, partiion=${consumerRecord.partition()}, offset=${consumerRecord.offset()}")
+                COUNT_KAFKA_CONSUMER_SYKETILFELLEBIT_TOMBSTONE.increment()
+                return
+            }
+            val syketilfelleBit: KafkaSyketilfellebit = configuredJacksonMapper()
```

Comment:
Mulig det ikke betyr så mye, men i stedet for å instansiere en ny mapper for hver record kunne man kanskje satt opp en egen deserializer i kafka-configen slik vi har gjort i en del av de andre appene?

Reply:
Jepp, jeg ordner egen deserializer 👍

Reply:
Fikset.